### ISSUE TITLE
Add messaging form on user and group profiles

### ISSUE THREAD

**Issue by michaltrzesimiech** [rtcharity/eahub.org#939]
Picking up where we left here https://github.com/rtcharity/eahub.org/issues/586

We need a messaging functionality. The scope is to make it possible for a logged in user to click a button "message" visible on other users and groups profiles and use a contact form to send an email i.e. using SendGrid (or whatever is good enough and less or equally as expensive). 

At the same time, we should disable the functionality to reveal users' email addresses publicly. 

There will be no functionality to receive replies through the Hub. Responding will happen through users' personal email accounts.

If economic, we should include some spam detection. If we find out this is complex task, we should move this to a separate ticket.

**Comment by sebgrebe** [comment]
@michaltrzesimiech My understanding is that @cafelow wanted to talk to CEA about this. 
My questions:
1) Were there any decisions made? Is there any input from CEA regarding this feature?
2) What is the cost situation on this? How many emails do we expect to be sent and would this get us over whatever threshold we have for what we are currently paying for with sendgrid?
3) @cafelow said that there were people who would like to contribute to this as devs. Should we get in touch with them/they with us before we start working on this?

**Comment by cafelow** [comment]
1. I talked to CEA's groups team, and also the tech team. I'm afraid this isn't in the tech team's priority right now, and isn't within the scope of the Groups team. To clarify, the groups team scope of helping the EA Hub is just the groups list + resources relating to groups. Since this feature isn't particularly useful for /groups (as they normally have a public email address), this isn't something the groups team will fund. They've suggested applying for a Meta grant if we want funding.

**Comment by vagarwalla** [comment]
**USER PROFILE WIREFRAME**
I created a wireframe here (ignore the profile redesign for now, will address in another Issue):
https://whimsical.com/5uUECYJe5vvWXZj5M9HDu8

In terms of User workflow:

1. User clicks "Message" button on Contact profile page
2. Message box pop-up appears
2. Optional: Message box autofills name/email 
3. User types in message
4. User clicks "Send"
5. An email is sent to the Contact
6. A confirmation email is sent to the User
7. User exits out of the message box pop-up using "Close" or the exit icon back to the Contact's page. 

Uncertainties

- Do we want to autofill the name/email or just let autofill do this?
- Do we want the message to be a pop-up or a new page? The design would stay basically the same


Does this look good or am I missing any important information?

**Comment by vagarwalla** [comment]
I've created issue #986 to update profile edit feautre so users can toggle messaging on/off.
